test/unit/org/apache/cassandra/nodes/NodesTest.java

```diff
@@ -260,6 +264,31 @@ public void testLocalInfoUnknownFieldsAreIgnoredDuringDeserialization() throws I
         assertEquals(clusterName, loadedLocalInfo.getClusterName());
     }
 
+    @Test
+    public void testPeerInfoFromDse6() throws IOException
+    {
+        InetAddressAndPort peer = InetAddressAndPort.getByName("127.0.0.2");
+        PeerInfo p = new PeerInfo(peer);
+        fakePeer(p);
+        TestDse6PeerInfo existingPeerInfo = new TestDse6PeerInfo(peer.address);
+        existingPeerInfo.setPreferredIp(p.getPreferred().address);
+        existingPeerInfo.setDataCenter(p.getDataCenter());
+        existingPeerInfo.setRack(p.getRack());
+        existingPeerInfo.setHostId(p.getHostId());
+
+        File local = dir.toPath().resolve("peers").toFile();
+        ObjectMapper objectMapper = Nodes.createObjectMapper();
+        SimpleModule module = new SimpleModule();
+        module.addSerializer(InetAddress.class, new InetAddressSerializer());
+        objectMapper.registerModule(module);
+        objectMapper.writerFor(new TypeReference<Collection<TestDse6PeerInfo>>() {}).writeValue(local, Collections.singleton(existingPeerInfo));
```

Comment:
@szymon-miezal 
we need deserialisation compatibility against DSE 6.8 metadata peers file.
specifically the PeerInfo class changed its fields from InetAddress to InetAddressAndPort.

i can't get this test to work yet, because the class name is part of the serialisation.  so i'm going to replace this^ and just put a sample file I've already got from DSE 6.8 into a location under `test/data/`